pkg/controller/rds/dbinstance/setup.go

```diff
@@ -656,16 +656,34 @@ func compareTimeRanges(format string, expectedWindow *string, actualWindow *stri
 	// all windows here have a "-" in between two values in the expected format, so just split
 	leftSpans := strings.Split(*expectedWindow, "-")
 	rightSpans := strings.Split(*actualWindow, "-")
+
+	if len(leftSpans) != 2 || len(rightSpans) != 2 {
+		return false, errors.New("invalid time window format")
+	}
+
 	for i := range leftSpans {
-		left, err := time.Parse(format, leftSpans[i])
+		// Extract day and time separately from the string before parsing
+		leftParts := strings.SplitN(leftSpans[i], ":", 2)
+		rightParts := strings.SplitN(rightSpans[i], ":", 2)
+
+		if len(leftParts) != 2 || len(rightParts) != 2 {
+			return false, errors.New("invalid time format, expected 'day:HH:MM'")
+		}
+
+		leftDay := strings.ToLower(strings.TrimSpace(leftParts[0]))
+		rightDay := strings.ToLower(strings.TrimSpace(rightParts[0]))
+
+		leftTime, err := time.Parse(format, leftSpans[i])
```

Comment:
This way we implement the parsing functionality twice, one for each value. Can we move this into an extra code block (can be in this package) with a parser function and on for equality check?

Might also make sense to do the parsing with a regex which is a bit more stable?

Reply:
Done. Moved parsing to a shared function and switched to regex-based parsing.